diff --git a/meson.build b/meson.build
index 174eb71c..5503c0be 100644
--- a/meson.build
+++ b/meson.build
@@ -202,7 +202,7 @@ endif
 
 bison = find_program('bison', required : false)
 flex = find_program('flex', 'lex', required : false)
-with_webalbums = bison.found() and flex.found()
+with_webalbums = false # bison.found() and flex.found()
 
 vector_ops = (c_comp.get_id() == 'gcc') and (c_comp.version().version_compare('>=4.0.7'))
 
