diff --git a/ext/libclementine-tagreader/tagreader.cpp b/ext/libclementine-tagreader/tagreader.cpp
index 42c9ce996..fb79ffeac 100644
--- a/ext/libclementine-tagreader/tagreader.cpp
+++ b/ext/libclementine-tagreader/tagreader.cpp
@@ -172,19 +172,7 @@ void TagReader::GuessArtistAndTitle(cpb::tagreader::SongMetadata* song) const {
 
 void TagReader::GuessAlbum(const QFileInfo& info,
                            cpb::tagreader::SongMetadata* song) const {
-  QString album = QString::fromStdString(song->album());
-  if (!album.isEmpty()) return;
-  const QString str_dir = info.absoluteDir().absolutePath();
-  if (str_dir.isEmpty()) return;
-  const QFileInfo dir(str_dir);
-  const QString dir_bn = dir.baseName();
-  if (dir_bn.isEmpty()) return;
-  album = ReplaceUnderscoresWithSpaces(dir_bn);
-  album = album.trimmed();
-  if (album.isEmpty()) return;
-  const QString al = album.toLower();
-  if (al == "various" || al == "downloads" || al == "music") return;
-  song->set_album(album.toUtf8().data());
+  return;
 }
 
 TagReader::TagReader()
