diff --git a/effects/desktopgrid/desktopgrid.cpp b/effects/desktopgrid/desktopgrid.cpp
index e7eb3b997..f80ea36f2 100644
--- a/effects/desktopgrid/desktopgrid.cpp
+++ b/effects/desktopgrid/desktopgrid.cpp
@@ -160,8 +160,10 @@ void DesktopGridEffect::prePaintScreen(ScreenPrePaintData& data, int time)
         }
         // PAINT_SCREEN_BACKGROUND_FIRST is needed because screen will be actually painted more than once,
         // so with normal screen painting second screen paint would erase parts of the first paint
-        if (timeline.currentValue() != 0 || (isUsingPresentWindows() && isMotionManagerMovingWindows()))
-            data.mask |= PAINT_SCREEN_TRANSFORMED | PAINT_SCREEN_BACKGROUND_FIRST;
+        if (timeline.currentValue() != 0 || (isUsingPresentWindows() && isMotionManagerMovingWindows())) {
+            const int paint = activated && timeline.currentValue() == 1 ? PAINT_SCREEN_TRANSFORMED : PAINT_SCREEN_REGION;
+            data.mask |= paint | PAINT_SCREEN_BACKGROUND_FIRST;
+        }
         if (!activated && timeline.currentValue() == 0 && !(isUsingPresentWindows() && isMotionManagerMovingWindows()))
             finish();
     }
