diff --git a/src/plugins/private/qml/WindowHeapDelegate.qml b/src/plugins/private/qml/WindowHeapDelegate.qml
index 795b92e3c6..8875b9d85f 100644
--- a/src/plugins/private/qml/WindowHeapDelegate.qml
+++ b/src/plugins/private/qml/WindowHeapDelegate.qml
@@ -250,7 +250,7 @@ ExpoCell {
             visible: !thumb.activeHidden && !activeDragHandler.active && !returnAnimation.running
             PC3.Label {
                 id: caption
-                visible: thumb.window.caption.length > 0 && thumb.windowTitleVisible
+                visible: false
                 width: thumb.width
                 maximumLineCount: 1
                 anchors.top: parent.bottom
