diff --git a/src/plugins/private/qml/WindowHeapDelegate.qml b/src/plugins/private/qml/WindowHeapDelegate.qml
index 795b92e3c6..1634fcce4f 100644
--- a/src/plugins/private/qml/WindowHeapDelegate.qml
+++ b/src/plugins/private/qml/WindowHeapDelegate.qml
@@ -358,31 +358,6 @@ ExpoCell {
             }
         }
 
-        Loader {
-            id: closeButton
-            LayoutMirroring.enabled: Qt.application.layoutDirection === Qt.RightToLeft
-
-            anchors {
-                right: thumbSource.right
-                top: thumbSource.top
-                margins: Kirigami.Units.smallSpacing
-            }
-            active: thumb.closeButtonVisible && (hoverHandler.hovered || Kirigami.Settings.tabletMode || Kirigami.Settings.hasTransientTouchInput) && thumb.window.closeable && !thumb.activeDragHandler.active && !returnAnimation.running
-
-            sourceComponent: PC3.Button {
-                text: i18ndc("kwin", "@info:tooltip as in: 'close this window'", "Close window")
-                icon.name: "window-close"
-                display: PC3.AbstractButton.IconOnly
-
-                PC3.ToolTip.text: text
-                PC3.ToolTip.visible: hovered && display === PC3.AbstractButton.IconOnly
-                PC3.ToolTip.delay: Kirigami.Units.toolTipDelay
-                Accessible.name: text
-
-                onClicked: thumb.window.closeWindow();
-            }
-        }
-
         Component.onDestruction: {
             if (selected) {
                 windowHeap.resetSelected();
