diff --git a/src/effects/desktopgrid/qml/main.qml b/src/effects/desktopgrid/qml/main.qml
index e1d8ac5da..802adb085 100644
--- a/src/effects/desktopgrid/qml/main.qml
+++ b/src/effects/desktopgrid/qml/main.qml
@@ -192,8 +192,8 @@ Rectangle {
 
         width: (parent.width + columnSpacing) * columns - columnSpacing
         height: (parent.height + rowSpacing) * rows - rowSpacing
-        rowSpacing: PlasmaCore.Units.gridUnit
-        columnSpacing: PlasmaCore.Units.gridUnit
+        rowSpacing: 0
+        columnSpacing: 0
         rows: container.effect.gridRows
         columns: container.effect.gridColumns
         transformOrigin: Item.TopLeft
