diff --git a/src/plugins/slide/slide.cpp b/src/plugins/slide/slide.cpp
index 7d0a71cba9..1755e458c9 100644
--- a/src/plugins/slide/slide.cpp
+++ b/src/plugins/slide/slide.cpp
@@ -93,8 +93,8 @@ void SlideEffect::reconfigure(ReconfigureFlags)
 
 void SlideEffectScreen::reconfigure()
 {
-    const qreal springConstant = 300.0 / effects->animationTimeFactor();
-    const qreal dampingRatio = 1.1;
+    const qreal springConstant = 500.0 / effects->animationTimeFactor();
+    const qreal dampingRatio = 1.05;
 
     m_motionX = SpringMotion(springConstant, dampingRatio);
     m_motionY = SpringMotion(springConstant, dampingRatio);
