diff --git a/src/plugins/slide/slide.cpp b/src/plugins/slide/slide.cpp
index 6808116f78..1a9d7d8fc4 100644
--- a/src/plugins/slide/slide.cpp
+++ b/src/plugins/slide/slide.cpp
@@ -63,8 +63,8 @@ void SlideEffect::reconfigure(ReconfigureFlags)
 {
     SlideConfig::self()->read();
 
-    const qreal springConstant = 300.0 / effects->animationTimeFactor();
-    const qreal dampingRatio = 1.1;
+    const qreal springConstant = 500.0 / effects->animationTimeFactor();
+    const qreal dampingRatio = 1.05;
 
     m_motionX = SpringMotion(springConstant, dampingRatio);
     m_motionY = SpringMotion(springConstant, dampingRatio);
