diff --git a/src/workspace.cpp b/src/workspace.cpp
index 1c0516e2b..44622ac61 100644
--- a/src/workspace.cpp
+++ b/src/workspace.cpp
@@ -1160,13 +1160,16 @@ void Workspace::updateCurrentActivity(const QString &new_activity)
     Window *window = nullptr;
 
     // FIXME below here is a lot of focuschain stuff, probably all wrong now
-    //  Keep active window focused if it's on the new activity
-    if (m_activeWindow && m_activeWindow->isShown() && m_activeWindow->isOnCurrentDesktop() && m_activeWindow->isOnCurrentActivity()) {
-        window = m_activeWindow;
-    } else if (options->focusPolicyIsReasonable()) {
+    if (options->focusPolicyIsReasonable()) {
         // Search in focus chain
         window = FocusChain::self()->getForActivation(VirtualDesktopManager::self()->currentDesktop());
     }
+    // If "unreasonable focus policy" and active_client is on_all_desktops and
+    // under mouse (Hence == old_active_client), conserve focus.
+    // (Thanks to Volker Schatz <V.Schatz at thphys.uni-heidelberg.de>)
+    else if (m_activeWindow && m_activeWindow->isShown() && m_activeWindow->isOnCurrentDesktop() && m_activeWindow->isOnCurrentActivity()) {
+        window = m_activeWindow;
+    }
 
     if (!window) {
         window = findDesktop(true, VirtualDesktopManager::self()->currentDesktop());
