diff --git a/daemon/powerdevilscreenbrightnesslogic.cpp b/daemon/powerdevilscreenbrightnesslogic.cpp
index 66e2a96..9e02c4e 100644
--- a/daemon/powerdevilscreenbrightnesslogic.cpp
+++ b/daemon/powerdevilscreenbrightnesslogic.cpp
@@ -40,6 +40,11 @@ int ScreenBrightnessLogic::calculateSteps(int maxValue) const
         return maxValue;
     }
 
+    // enable finer control for screens with higher max brightness values
+    if (maxValue >= 800) {
+        return 100;
+    }
+
     if (maxValue >= 100 || maxValue % 20 == 0 || (maxValue >= 80 && maxValue % 4 == 0)) {
         // In this case all 20 steps are perfect.
         return 20;
