Index: ./src/VBox/Runtime/common/fs/ntfsvfs.cpp
===================================================================
--- ./src/VBox/Runtime/common/fs/ntfsvfs.cpp	(revision 134297)
+++ ./src/VBox/Runtime/common/fs/ntfsvfs.cpp	(working copy)
@@ -1052,7 +1052,7 @@
                     case 4: cClustersInRun |= (uint32_t)pbPairs[offPairs + 3] << 24; RT_FALL_THRU();
                     case 3: cClustersInRun |= (uint32_t)pbPairs[offPairs + 2] << 16; RT_FALL_THRU();
                     case 2: cClustersInRun |= (uint16_t)pbPairs[offPairs + 1] <<  8; RT_FALL_THRU();
-                    case 1: cClustersInRun |= (uint16_t)pbPairs[offPairs + 0] <<  0; RT_FALL_THRU();
+                    case 1: cClustersInRun |= (uint16_t)pbPairs[offPairs + 0] <<  0;
                 }
                 offPairs += cbRunField;
                 AssertBreakStmt(cClustersInRun <= cMaxClustersInRun,
