diff --git a/wallpapers/image/plugin/imagebackend.cpp b/wallpapers/image/plugin/imagebackend.cpp
index 1863b1142a..855457b20f 100644
--- a/wallpapers/image/plugin/imagebackend.cpp
+++ b/wallpapers/image/plugin/imagebackend.cpp
@@ -416,7 +416,7 @@ void ImageBackend::nextSlide()
     QString next = m_slideFilterModel->index(m_currentSlide, 0).data(ImageRoles::PackageNameRole).toString();
     // And  avoid showing the same picture twice
     if (previousSlide == rowCount - 1 && previousPath == next && rowCount > 1) {
-        m_currentSlide += 1;
+        m_slideFilterModel->swapFirstWithRandom();
         next = m_slideFilterModel->index(m_currentSlide, 0).data(ImageRoles::PackageNameRole).toString();
     }
     if (next.isEmpty()) {
diff --git a/wallpapers/image/plugin/slidefiltermodel.cpp b/wallpapers/image/plugin/slidefiltermodel.cpp
index ed10a20b33..29b1326f1f 100644
--- a/wallpapers/image/plugin/slidefiltermodel.cpp
+++ b/wallpapers/image/plugin/slidefiltermodel.cpp
@@ -211,6 +211,26 @@ void SlideFilterModel::invalidateFilter()
     QSortFilterProxyModel::invalidateFilter();
 }
 
+void SlideFilterModel::swapFirstWithRandom()
+{
+    const int size = rowCount();
+    if (size < 2)
+        return;
+
+    // [start, end)
+    const int pos = QRandomGenerator::global()->bounded(1, size);
+    // get the randomOrder index of the sourceModel index of the proxy index
+    const QModelIndex srcIdxA = mapToSource(index(0, 0));
+    const QModelIndex srcIdxB = mapToSource(index(pos, 0));
+    const int rndIdxA = m_randomOrder.indexOf(srcIdxA.row());
+    const int rndIdxB = m_randomOrder.indexOf(srcIdxB.row());
+    // and finally swap
+    m_randomOrder.swapItemsAt(rndIdxA, rndIdxB);
+
+    QSortFilterProxyModel::invalidate();
+    sort(0);
+}
+
 int SlideFilterModel::indexOf(const QString &path)
 {
     if (!sourceModel())
diff --git a/wallpapers/image/plugin/slidefiltermodel.h b/wallpapers/image/plugin/slidefiltermodel.h
index 8fa65b2307..c0f34cedca 100644
--- a/wallpapers/image/plugin/slidefiltermodel.h
+++ b/wallpapers/image/plugin/slidefiltermodel.h
@@ -31,6 +31,7 @@ public:
     void setSourceModel(QAbstractItemModel *sourceModel) override;
     void invalidate();
     void invalidateFilter();
+    void swapFirstWithRandom();
 
     Q_INVOKABLE int indexOf(const QString &path);
     Q_INVOKABLE void openContainingFolder(int rowIndex);
