diff --git a/applets/systemmonitor/systemmonitor/systemmonitor.cpp b/applets/systemmonitor/systemmonitor/systemmonitor.cpp
index 9b60364dee..a76a85bafb 100644
--- a/applets/systemmonitor/systemmonitor/systemmonitor.cpp
+++ b/applets/systemmonitor/systemmonitor/systemmonitor.cpp
@@ -7,7 +7,6 @@
 #include "systemmonitor.h"
 
 #include <QDebug>
-#include <QQmlEngine>
 #include <QQuickItem>
 #include <QQuickWindow>
 #include <QStandardPaths>
@@ -37,7 +36,7 @@ void SystemMonitor::init()
     // FIXME HACK: better way to get the engine At least AppletQuickItem should have an engine() getter
     auto qmlObject = new PlasmaQuick::SharedQmlEngine();
     KConfigGroup cg = config();
-    m_sensorFaceController = new KSysGuard::SensorFaceController(cg, qmlObject->engine().get(), new QQmlEngine(this));
+    m_sensorFaceController = new KSysGuard::SensorFaceController(cg, qmlObject->engine().get());
     qmlObject->deleteLater();
 
     // NOTE: taking the pluginId from the child applet (cpu monitor, memory, whatever) is done implicitly by not embedding metadata in this applet
@@ -55,7 +54,7 @@ KSysGuard::SensorFaceController *SystemMonitor::faceController() const
 KSysGuard::SensorFaceController *SystemMonitor::workaroundController(QQuickItem *context) const
 {
     KConfigGroup cg = config();
-    return new KSysGuard::SensorFaceController(cg, qmlEngine(context), new QQmlEngine(context));
+    return new KSysGuard::SensorFaceController(cg, qmlEngine(context));
 }
 
 void SystemMonitor::configChanged()
