commit 8ece96215b1b2b520e2f4fdd16922179f5a0e7bf
Author: Sebastian Meyer <mail@bastimeyer.de>
Date:   Mon Sep 2 10:13:44 2024 +0200

    wallpapers/image: strict slideshow intervals

diff --git a/wallpapers/image/plugin/imagebackend.cpp b/wallpapers/image/plugin/imagebackend.cpp
index ec9a52a383..9fb9f365f5 100644
--- a/wallpapers/image/plugin/imagebackend.cpp
+++ b/wallpapers/image/plugin/imagebackend.cpp
@@ -14,6 +14,7 @@
 
 #include <math.h>
 
+#include <QDateTime>
 #include <QDir>
 #include <QFileInfo>
 #include <QGuiApplication>
@@ -33,6 +34,8 @@ ImageBackend::ImageBackend(QObject *parent)
     : QObject(parent)
     , m_targetSize(qGuiApp->primaryScreen()->size() * qGuiApp->primaryScreen()->devicePixelRatio())
 {
+    m_timer.setSingleShot(true);
+    m_timer.setTimerType(Qt::PreciseTimer);
     connect(&m_timer, &QTimer::timeout, this, &ImageBackend::nextSlide);
 }
 
@@ -399,8 +402,6 @@ void ImageBackend::nextSlide()
         m_currentSlide += 1;
         next = m_slideFilterModel->index(m_currentSlide, 0).data(ImageRoles::PackageNameRole).toString();
     }
-    m_timer.stop();
-    m_timer.start(m_delay * 1000);
     if (next.isEmpty()) {
         m_image = QUrl::fromLocalFile(previousPath);
     } else {
@@ -409,6 +410,26 @@ void ImageBackend::nextSlide()
     }
 
     saveCurrentWallpaper();
+
+    // Synchronize wallpaper transitions across desktops on all screens and activities
+    // by updating the wallpaper in strict time intervals since UNIX epoch, defined by `m_delay`.
+    QDateTime dtnow = QDateTime::currentDateTimeUtc();
+    qint64 now = dtnow.toMSecsSinceEpoch();
+    int interval_ms = m_delay * 1000;
+    int offset_ms = now % interval_ms;
+    int duration_ms = interval_ms - offset_ms;
+
+    // Avoid overlap of wallpaper transition animations by adding a whole interval if `duration_ms` is too short:
+    // 1. Despite `m_timer` being a `Qt::PreciseTimer`, it can still sometimes trigger a bit too early,
+    //    causing `duration_ms` to be only a couple of milliseconds.
+    // 2. The slideshow could have been started right before the end of the current interval,
+    //    resulting in the same issue.
+    if (duration_ms < m_timer_duration_min) {
+        duration_ms += interval_ms;
+    }
+
+    // Always add a tiny offset, as `Qt::PreciseTimer` timers are actually not precise.
+    m_timer.start(duration_ms + m_timer_duration_offset);
 }
 
 void ImageBackend::slotSlideModelDataChanged(const QModelIndex &topLeft, const QModelIndex &bottomRight, const QList<int> &roles)
diff --git a/wallpapers/image/plugin/imagebackend.h b/wallpapers/image/plugin/imagebackend.h
index 57f65da94b..84b39de8ab 100644
--- a/wallpapers/image/plugin/imagebackend.h
+++ b/wallpapers/image/plugin/imagebackend.h
@@ -183,6 +183,8 @@ private:
     Q_OBJECT_BINDABLE_PROPERTY_WITH_ARGS(ImageBackend, bool, m_slideshowFoldersFirst, false, &ImageBackend::slideshowFoldersFirstChanged)
 
     QTimer m_timer;
+    static constexpr int m_timer_duration_min = 1000;
+    static constexpr int m_timer_duration_offset = 10;
     bool m_pauseSlideshow = false;
     std::optional<decltype(std::declval<QTimer>().remainingTimeAsDuration())> m_remainingTime;
 
